### PR TITLE
🏗 Exempt `build-system/` and `test/` from JSDoc requirements

### PR DESCRIPTION
**Pull request by rsimha**
This PR makes #15977 less painful to deal with.

### CODE REVIEW

File: test/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "rules": {
+    "require-jsdoc": 0
```
Comment (choumx) [inline test/.eslintrc]: Won't cover extension tests. Is it possible to do this with file glob? E.g. `test-*.js`.

Comment (rsimha) [reply test/.eslintrc]: Yeah, I was thinking of that. Not sure it's possible to disable specific rules for specific directories that match a pattern.

One way to do so would be to entirely disable linting for extension test directories, but that's bad.

I'll look for a way to do this and check it in separately.

Comment (rsimha) [reply test/.eslintrc]: Done! Updated commit pushed. Tested locally.

**Pull request merged**